lib/ace/snippets.js

```diff
@@ -925,7 +924,6 @@ var TabstopManager = function(editor) {
                 range.original = p;
                 range.tabstop = dest;
                 ranges.push(range);
-                ts.rangeList.ranges.push(range);
                 if (dest != ts)
```

Comment:
I'm not sure if I understand correctly, but can you replace this 
```
if (dest != ts)
  dest.unshift(range);
else
  dest[i] = range;
```

with `dest.push(range)`, since you call `addList` later on anyways that does the sorting?

Reply:
addList sorts ranges in rangeList, it doesn't modify dest

Reply:
Sure, but do you even still need the ranges in dest if they're also sorted in dest.rangeList?

Reply:
We need the original order of ranges, to make sure that the primary selection doesn't change when expanding snippets from multiple cursors, it may be possible to remove ranges in dest and only keep the primary range, but not sure if it will be much simpler.